tracer/encoder.go

```diff
@@ -119,62 +97,41 @@ func (e *jsonEncoder) ContentType() string {
 	return e.contentType
 }
 
-func (e *jsonEncoder) Buffer() *bytes.Buffer {
-	return e.buffer
-}
-
-func (e *jsonEncoder) SetBuffer(b *bytes.Buffer) {
-	e.buffer = b
-}
-
 const (
-	JSON_ENCODER = iota
-	MSGPACK_ENCODER
+	jsonType = iota
+	msgpackType
```

Comment:
I don't really see the point in having an int to describe the encoder type here. Just use the encoding as the key. Let the constants be `jsonType =  "application/json"` and `msgpackType = "application/msgpack"`, and use those directly. Using an int is not like using a value within a finite set so we always take the risk to pass `-1` or `2`, get out of bound, and return `nil` when calling `contentType(int)`. So let's just use the encoding as the key, but still use the constants `jsonType` and `msgpackType` to avoid copy/pasting and typo errors.